### PR TITLE
Gonix Tests suite

### PR DESCRIPTION
**Pull request by vanhalt**
Initial tests

- created tests/ dir
- added gonix_test.go file
- Testing:
  - basename
  - dirname
  - echo
  - false
  - true

In this commit *dirname* tests are failing, when passing `/tmp/` should return `/` instead of `/tmp`

### CODE REVIEW

**Comment by polegone** [comment]
A test suite is a great idea.

File: Makefile [hunk 1]
```diff
@@ -11,6 +11,9 @@ GOFLAGS=
 
 all: build basename cat cp dirname echo false head mkdir nl pwd seq sleep tail tee touch true xxd yes
 
+testing:
```
Comment (mibk) [inline Makefile]: I suggest to rename `testing` to `test`. But more importantly, this target should either
1. depend on *all target* (`test: all`) as binaries may not exist
2. or `exec.Command` `go run $filename` rather then `build/$binaryName`.

Using `gofmt` before commit is also a best practice...

**Comment by vanhalt** [comment]
Thank you! Didn't know how to actually write that, the make test, now I do. About the gofmt, you're totally right... I am still getting use to coding in Go. Great feedback

:D

**Comment by mibk** [comment]
I'm glad it helped :smiley:

**Pull request merged**